### PR TITLE
UXD-302 Button Documentation mdx

### PR DESCRIPTION
**Pull request by KDarcey**
### Purpose 🚀
To create documentation for the Button component to improve user experience for consumers.

### Notes ✏️
N/A

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [ ] MINOR (backward compatible) change to _these packages_
- [x] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-302-Button-documentation-mdx

### Screenshots 📸
N/A

### References 🔗
https://aclgrc.atlassian.net/browse/UXD-302

### CODE REVIEW

File: packages/Button/stories/Button.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    isToolshown: true,
+    showPanel: false,
+  },
+};
+
+<Meta title={`${getStoryName("Button")}/Docs`} component={Button} parameters={params} />
+
+<Description markdown={Readme} />
+
+<MDX />
```
Comment (mikrotron) [inline packages/Button/stories/Button.stories.mdx:19]: Since this `Button.stories.mdx` file is now so small, how about we just add the content of `Button.mdx` right here to simplify things?

File: packages/Button/stories/Button.mdx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+import ButtonSubmit from "./examples/ButtonSubmit";
+import CloseButtonRef from "./examples/CloseButtonRef";
+import Focus from "./examples/Focus";
+import NewRef from "./examples/NewRef";
+import OldRef from "./examples/OldRef";
+import Button from "../src/";
+
+## More Examples
+
+### Index
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:13]: This is awesome 👍

File: packages/Button/stories/Button.mdx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+To add a basic button import the button and add the `<Button>` component to your project.
+
+```js
+<Button>Click Me!</Button>
+```
+
+##### Result
+
+<Button>Click Me!</Button>
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:39]: I think this is redundant – we have a real basic Usage example above in the README. Lets go straight into the Kinds example.

File: packages/Button/stories/Button.mdx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+#### Kinds
+
+Differant projects require differant buttons, luckily the Button component can make us of the `kind` property to create a variety of buttons for differant use cases.
+
+```js
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:45]: I find the `jsx` syntax highlighting more pleasing 😁

File: packages/Button/stories/Button.mdx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+##### Result
+
+<Button kind="default">default</Button>
+<Button kind="primary">primary</Button>
+<Button kind="secondary">secondary</Button>
+<Button kind="destructive">destructive</Button>
+<Button kind="flat">flat</Button>
+<Button kind="minor">minor</Button>
+<Button kind="link">link</Button>
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:63]: I think it would be nice two wrap these results with a bit of padding.  

At least a `<p>`, but you could also create a new helper in `.storybook/assets/common.styles.js` that has a bit more bottom margin to space it out from the following example.  

Or maybe just use the `<ButtonStory>` component?

File: packages/Button/stories/Button.mdx [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+<Button isFullWidth>full width</Button>
+
+#### States
+
+Buttons can be display differant states by adjusting making use of the `isActive`, `isPending`, and `isDisabled` properties.
+
+```js
+    <Button isActive>active</Button>
+    <Button isDisabled>disabled</Button>
+    <Button isPending>pending</Button>
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:90]:
```suggestion
<Button isActive>active</Button>
<Button isDisabled>disabled</Button>
<Button isPending>pending</Button>
```

File: packages/Button/stories/Button.mdx [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+To turn a Button into a submit Button by adding the `isSubmit` property.
+
+```js
+<form
+  onSubmit={event => {
+    alert("submitted");
+    event.preventDefault();
+  }}
+>
+  <input type="text" defaultValue="press enter while focus" />
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:110]: It's so ugly 🙈  

The Paprika `<Input>` would be much nicer, but you'd have to change the `<ButtonSubmit>` story too, and add some custom CSS... 😓

**Comment by mikrotron** [comment]
There's this weird bit in the README that seems out of place.  Can you remove it?
https://github.com/acl-services/paprika/blob/master/packages/Button/README.md#buttonclose-1

File: packages/Button/stories/Button.mdx [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+<Button.Link href="https://youtu.be/IdkCEioCp24?t=92" shouldOpenNewTab>
+    Link in new tab
+</Button.Link>
+<Button.Link kind="secondary" icon={<PlusIcon />} href="https://youtu.be/IdkCEioCp24?t=92" shouldOpenNewTab>
+    Button link with icon
+</Button.Link>
+
+### Icon Button
+
+The Icon Button allows users to easily make use of buttons that contain icons or use icons as buttons.
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:188]:
```suggestion
To make use of an Icon Button use the `<Button.Icon>` element. The Icon Button allows users to easily render a square button that uses _only_ an icon as the label.

For accessibility, it is important to also provide a text label for Icon Buttons by utilizing the `a11yText` prop.
```

File: packages/Button/stories/Button.mdx [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+    <InfoIcon />
+</Button.Icon>
+<Button.Icon  kind="minor">
+    <InfoIcon />
+</Button.Icon>
+<Button.Icon  icon="🐐" kind="secondary" size="small">
+    <span role="img" aria-label="Llama">
+    🦙
+    </span>
+</Button.Icon>
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:210]: Let's cut this one ✂️ 
It's sort of a silly edge case test, not how we _should_ use the component.

File: packages/Button/stories/Button.mdx [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+</p>
+```
+
+##### Result
+
+<CloseButtonRef />
+
+### Link Button
+
+To make use of a Link Button use the `<Link.Button>` element.
```
Comment (mikrotron) [inline packages/Button/stories/Button.mdx:140]:
```suggestion
To make use of a Link Button use the `<Button.Link>` element.
```

**Comment by mikrotron** [comment]
What do you think about adding this:

```scss

// DOCS

// Props Tables
.sbdocs-table {
  width: 100%;
  
  th:last-child {
    width: 50%;
  }
}
```

To the bottom of `.storybook/reset.scss`?

cc @oscarkwan

**Review by mikrotron — APPROVED** [review]
llgtm 100
